classes/app.rb

```diff
@@ -0,0 +1,141 @@
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+  end
+
+  def run
+    actions.each { |key, value| puts "#{key}: #{value}" }
+    action = gets.chomp.to_i
+    action(action)
+  end
+
+  private
+
+  def actions
+    puts # Add a blank line before the actions list
+    {
+      1 => 'List all books',
+      2 => 'List all people',
+      3 => 'Create a person',
+      4 => 'Create a book',
+      5 => 'Create a rental',
+      6 => 'List all rentals for a given person id',
+      7 => 'Exit'
+    }
+  end
+
+  def action(action)
+    action_list = { 1 => 'list_books', 2 => 'list_people', 3 => 'add_person', 4 => 'add_book', 5 => 'add_rental',
+                    6 => 'list_rentals_by_person_id', 7 => 'exit' }
+
+    if action_list.key?(action)
+      send(action_list[action])
+    else
+      puts 'Invalid input!'
+    end
+
+    run # List actions once the user is done with the current action
```

Comment:
- [ ] Kindly move this highlighted function action to main.rb, like the previous reviewer requested. It will ensure the separation of functionalities between the user inputs and data processing.

Reply:
Moves action function to main.rb